crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -387,6 +383,82 @@ impl QueryBuilder {
         Ok(query)
     }
 
+    fn multi_get_events<'a>(
+        cursor: Option<(i64, i64)>,
+        descending_order: bool,
+        limit: i64,
+        filter: Option<EventFilter>,
+    ) -> Result<events::BoxedQuery<'a, Pg>, Error> {
+        let mut query = events::dsl::events.into_boxed();
+        if let Some(cursor_val) = cursor {
+            if descending_order {
+                query = query.filter(events::dsl::tx_sequence_number.lt(cursor_val.0));
+                query = query.filter(events::dsl::event_sequence_number.lt(cursor_val.1));
+            } else {
+                query = query.filter(events::dsl::tx_sequence_number.gt(cursor_val.0));
+                query = query.filter(events::dsl::event_sequence_number.gt(cursor_val.1));
+            }
```

Comment:
This seems wrong.  Imagine I have the following transactions and events:

- `TX0(E0, E1, E2)`
- `TX1(E0, E1)`
- `TX2(E0, E1)`

And the cursor was on `TX0.E0` -- then this query, which filters by both the transaction sequence number and the event sequence number, will no include `TX1.E0` or `TX2.E0`.

Reply:
Ah.... yeah thanks for catching this

For descending order:
1. Fetch events from transactions with a lower tx_sequence_number than the cursor.
2. OR, if same tx_sequence_number, fetch events with a lower event_sequence_number

For ascending order:

1. Fetch events from txns with higher tx_sequence_number than cursor
2. OR, if same tx_sequence_number, fetch events with a higher event_sequence_number